x/incentive/client/tests/tests.go

```diff
@@ -0,0 +1,234 @@
+				Bonded: sdk.NewCoins(),
+			},
+		},
+		{
+			Name:     "query total unbonding - no denom",
+			Command:  cli.GetCmdQueryTotalUnbonding(),
+			Args:     []string{},
+			Response: &incentive.QueryTotalUnbondingResponse{},
+			ExpectedResponse: &incentive.QueryTotalUnbondingResponse{
+				Unbonding: sdk.NewCoins(),
+			},
+		},
+		{
+			Name:    "query total unbonding - specific denom",
+			Command: cli.GetCmdQueryTotalUnbonding(),
+			Args: []string{
+				"u/" + appparams.BondDenom,
+			},
+			Response: &incentive.QueryTotalUnbondingResponse{},
+			ExpectedResponse: &incentive.QueryTotalUnbondingResponse{
+				Unbonding: sdk.NewCoins(),
+			},
+		},
+		{
+			Name:    "query current rates (zero)",
+			Command: cli.GetCmdQueryCurrentRates(),
+			Args: []string{
+				"u/" + appparams.BondDenom,
+			},
+			Response: &incentive.QueryCurrentRatesResponse{},
+			ExpectedResponse: &incentive.QueryCurrentRatesResponse{
+				ReferenceBond: sdk.NewInt64Coin("u/"+appparams.BondDenom, 1),
+				Rewards:       sdk.NewCoins(),
+			},
+			ErrMsg: "",
+		},
+		{
+			Name:    "query actual rates (zero)",
+			Command: cli.GetCmdQueryActualRates(),
+			Args: []string{
+				"u/" + appparams.BondDenom,
+			},
+			Response: &incentive.QueryActualRatesResponse{},
+			ExpectedResponse: &incentive.QueryActualRatesResponse{
+				APY: sdk.ZeroDec(),
+			},
+			ErrMsg: "",
+		},
+		{
+			Name:    "query pending rewards (zero)",
+			Command: cli.GetCmdQueryPendingRewards(),
+			Args: []string{
+				val.Address.String(),
+			},
+			Response: &incentive.QueryPendingRewardsResponse{},
+			ExpectedResponse: &incentive.QueryPendingRewardsResponse{
+				Rewards: sdk.NewCoins(),
+			},
+			ErrMsg: "",
+		},
+	}
+
+	supplyCollateral := itestsuite.TestTransaction{
+		Name:    "(setup) supply collateral",
+		Command: leveragecli.GetCmdSupplyCollateral(),
+		Args: []string{
+			"300uumee",
+		},
+		ExpectedErr: nil,
+	}
+
+	bond := itestsuite.TestTransaction{
+		Name:    "bond",
+		Command: cli.GetCmdBond(),
+		Args: []string{
+			"300u/uumee",
+		},
+		ExpectedErr: nil,
+	}
+
+	beingUnbonding := itestsuite.TestTransaction{
+		Name:    "begin unbonding",
+		Command: cli.GetCmdBeginUnbonding(),
+		Args: []string{
+			"100u/uumee",
+		},
+		ExpectedErr: nil,
+	}
+
+	emergencyUnbond := itestsuite.TestTransaction{
+		Name:    "emergency unbond",
+		Command: cli.GetCmdEmergencyUnbond(),
+		Args: []string{
+			"100u/uumee",
+		},
+		ExpectedErr: nil,
+	}
+
+	claim := itestsuite.TestTransaction{
+		Name:        "claim",
+		Command:     cli.GetCmdClaim(),
+		Args:        []string{},
+		ExpectedErr: nil,
+	}
+
+	sponsor := itestsuite.TestTransaction{
+		Name:        "sponsor (program does not exist)",
+		Command:     cli.GetCmdSponsor(),
+		Args:        []string{"1"},
+		ExpectedErr: sdkerrors.ErrNotFound,
+	}
+
+	s.RunTestQueries(zeroQueries...)
+
+	s.RunTestTransactions(
+		supplyCollateral,
+		bond,
+		beingUnbonding,
+		emergencyUnbond,
+		claim,
+		sponsor,
+	)
```

Comment:
shall we run some queries after the transactions?

Reply:
for now, we can skip that.

There are minimal tests designed to ensure the commands worked, rather than scenarios where the results of these actions are being verified (that happens in msg_server_test and scenario_test since they run faster.)